src/sysupdate/sysupdated.c

```diff
@@ -0,0 +1,1857 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "sd-bus.h"
+
+#include "build-path.h"
+#include "bus-error.h"
+#include "bus-get-properties.h"
+#include "bus-label.h"
+#include "bus-log-control-api.h"
+#include "bus-polkit.h"
+#include "common-signal.h"
+#include "discover-image.h"
+#include "env-util.h"
+#include "event-util.h"
+#include "escape.h"
+#include "fd-util.h"
+#include "fileio.h"
+#include "main-func.h"
+#include "memfd-util.h"
+#include "mkdir-label.h"
+#include "os-util.h"
+#include "process-util.h"
+#include "sd-json.h"
+#include "service-util.h"
+#include "signal-util.h"
+#include "socket-util.h"
+#include "string-table.h"
+
+typedef struct Manager {
+        sd_event *event;
+        sd_bus *bus;
+
+        Hashmap *targets;
+
+        uint64_t current_job_id;
+        Hashmap *jobs;
+
+        Hashmap *polkit_registry;
+
+        sd_event_source *notify_event;
+} Manager;
+
+typedef enum TargetClass {
+        /* These should try to match ImageClass from src/basic/os-util.h */
+        TARGET_MACHINE  = IMAGE_MACHINE,
+        TARGET_PORTABLE = IMAGE_PORTABLE,
+        TARGET_SYSEXT   = IMAGE_SYSEXT,
+        TARGET_CONFEXT  = IMAGE_CONFEXT,
```

Comment:
idea: also define a pseudo feild here:

```
…
_TARGET_CLASS_IS_IMAGE_CLASS_MAX,
TARGET_HOST = _TARGET_CLASS_IS_IMAGE_CLASS_MAX,
…
```

And then use that for the assert_cc:

```c
assert_cc(_TARGET_CLASS_IS_IMAGE_CLASS_MAX == _IMAGE_CLASS_MAX);
```

but doesn't really matter

Reply:
I had to disable a warning for comparing enums of different types to make this work, but I suppose that's to be expected.